paasta_tools/cli/cmds/secret.py

```diff
@@ -269,12 +422,60 @@ def paasta_secret(args):
 
         print_paasta_helper(secret_path, args.secret_name, args.shared)
     elif args.action == "decrypt":
+        secret_provider = _get_secret_provider_for_service(
+            service,
+            cluster_names=args.clusters,
+            # `decrypt` does not require the current working directory
+            # to be a writeable git checkout of yelpsoa-configs
+            soa_dir=args.yelpsoa_config_root,
+        )
         print(
             decrypt_secret(
                 secret_provider=secret_provider, secret_name=args.secret_name
             ),
             end="",
         )
+    elif args.action == "run":
+        new_environ = os.environ.copy()
+
+        system_paasta_config = load_system_paasta_config()
+        secret_provider_kwargs = {
+            "vault_cluster_config": system_paasta_config.get_vault_cluster_config(),
+            "vault_auth_method": args.vault_auth_method,
+            "vault_token_file": args.vault_token_file,
+        }
+
+        # This includes only the environment variables mapped to secrets,
+        # other environment variables are not included.
+        # All environment variables set in the current shell will also
+        # be passed through.
+        new_secret_vars = decrypt_secret_environment_variables(
+            secret_provider_name=system_paasta_config.get_secret_provider_name(),
+            environment=get_instance_config(
+                service=args.service,
+                instance=args.instance,
+                cluster=args.clusters,
+            ).get_env(),
+            soa_dir=args.yelpsoa_config_root,
+            service_name=args.service,
+            cluster_name=args.clusters,
+            secret_provider_kwargs=secret_provider_kwargs,
+        )
+
+        for var_name, var_value in new_secret_vars.items():
+            new_environ[var_name] = var_value
+
+        if args.cmd == ["bash"]:
+            # make it clear that we're running in a sub-shell
+            new_environ["PS1"] = r"(paasta secret run) \$ "
```

Comment:
tbh i wasn't sure if people would want their PS1 messed with at all, this was mostly to prevent people from running it without a command and thinking it exited without doing anything